### PR TITLE
[DS-3967] 6x - Migrate Dockerfile to DSpace/DSpace

### PR DESCRIPTION
**Pull request by terrywbrady**
https://jira.duraspace.org/browse/DS-3967

### CODE REVIEW

File: docker.local.cfg [hunk 1]
```diff
@@ -0,0 +1,4 @@
+dspace.dir = /dspace
```
Comment (tdonohue) [inline docker.local.cfg]: As noted in today's meeting, let's move this to `[src]/src/main/docker/local.cfg`

Can we also add some comments at the top of this file to note that this is specific to Docker and used by the DockerFile (or something like that)?

**Comment by ppmdo** [comment]
The local.cfg can be gerated at build time as shown here: https://github.com/ppmdo/dspace-docker/blob/93bb794e97b9fa90ed00981721b3de809f1cf5f5/Dockerfile#L52

Although I think that approach can get quite complicated if the dspace.cfg many modifications.

Also, It could be useful to build with Mirage2 included and the let the user select what theme to use via an ENV variable as shown here: https://github.com/ppmdo/dspace-docker/blob/93bb794e97b9fa90ed00981721b3de809f1cf5f5/Dockerfile#L63

**Comment by terrywbrady** [comment]
@ppmdo , thanks for the comments.  I had not thought about Mirage2 support.

I am going to test building the images with Mirage2 enabled by default.  It will slow the build down a bit, but it should give the end user more flexibility with theme selection.  I will share my progress on this.

I envision that the user will use a volume mount in the docker compose file to override any additional properties at runtime (local.cfg, xmlui.xconf).  My hope is to minimize the number of variant Docker images that the project will support.

Now that the Dockerfile will exist on each supported branch of DSpace, I have customized the Dockerfiles to provide a minimal local.cfg or build.properties file within each branch.  If you take a look at the related PR's you will see that I am providing the files to be overridden.

- https://github.com/DSpace/DSpace/pull/2137 - 4x (build.properties)
- https://github.com/DSpace/DSpace/pull/2136 - 5x (build.properties)
- https://github.com/DSpace/DSpace/pull/2135 - 7x (local.cfg)

**Comment by ppmdo** [comment]
I agree with the files being overridden instead of generated. It's makes it simpler yet flexible.

**Comment by ppmdo** [comment]
Tested on Fedora 27. Works flawlessly.

**Comment by ppmdo** [comment]
Tested in Windows 10 with the following docker-compose file: https://github.com/DSpace-Labs/DSpace-Docker-Images/pull/51/files#diff-e437aab83f63209d609c14191f498e5e

Works without problems, with and without the Mirage2 option.

**Review by tdonohue — APPROVED** [review]
👍 I haven't tested, but the Dockerfile & comments within make sense.  As @ppmdo has tested this in several scenarios (see above comments), I think this is ready to merge.

**Pull request merged**